### PR TITLE
Restoring `azl` macro usage.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Re-introducing the changes from #8232. They were breaking toolchain builds and had to be reverted in #8279. This PR restores the use of the `azl` macro but also addresses the toolchain build issues related to it.

###### Change Log  

- Restored the use of the `azl` macro to define major version number for specs tied to the system version.
- Updated toolchain builds to consistently use pass the `azl` macro into its RPM queries and build commands.

###### Does this affect the toolchain?  

No.

###### Associated issues  

- #8232
- #8279

###### Test Methodology

- [Full dev build](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=522924&view=results).

### CODE REVIEW

File: SPECS/azurelinux-repos/azurelinux-repos.spec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Summary:        AzureLinux repo files, gpg keys
 Name:           azurelinux-repos
-Version:        3.0
+Version:        %{azl}.0
```
Comment (mfrw) [inline SPECS/azurelinux-repos/azurelinux-repos.spec:3]: double checking here:
Does our `autoupgrade` tooling work with this change ?

**Review by mfrw — COMMENTED** [review]
A couple more package I feel we should include here:
- `core-packages`
- `asc`
- `build-essential`
- `distroless-packages`
- `azurelinux-repos`
- `prebuilt-ca-certificates-base` (maybe)

File: SPECS/azurelinux-repos/azurelinux-repos.spec (hunk 1 above)
Comment (PawelWMS) [reply SPECS/azurelinux-repos/azurelinux-repos.spec:3]: I don't think so, but I believe the auto-upgrades are for cases where we can get a newer spec from another distro. The specs modified in this PR are the ones we either created ourselves or they moved away from the original source so much (for instance due to a name change), that an auto-upgrade wouldn't work anyway.

**Comment by PawelWMS** [comment]
> A couple more package I feel we should include here:
> 
> * `core-packages`
> * `asc`
> * `build-essential`
> * `distroless-packages`
> * `azurelinux-repos`
> * `prebuilt-ca-certificates-base` (maybe)

Good idea, added the missing ones.

**Review by mfrw — APPROVED** [review]
LGTM :)

**Review by nicogbg — APPROVED** [review]

**Review by anphel31 — APPROVED** [review]

**Pull request merged**